pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -424,6 +424,7 @@ func (m *rangefeedMuxer) startNodeMuxRangeFeed(
 			for _, s := range toRestart {
 				s.release()
 			}
+			m.metrics.Errors.RangefeedFatalErrors.Inc(int64(len(toRestart)))
 			return err
 		}
 
```

Comment:
Not part of your PR but the existing code is causing complexity.`restartActiveRangeFeeds` has an early return when restarting rangefeeds, so we might be missing some of the `RangefeedFatalErrors` for rangefeeds that didn't make to `restartActiveRangeFeed`. We probably shouldn't allow `restartActiveRangeFeeds` to early return since it also holds onto the other resources that need to be released from `s.release()`. 

https://github.com/cockroachdb/cockroach/blob/ffe386c9b758a2ede1126959be4a1bf2726a1a54/pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go#L516-L525

Reply:
I'm kinda wondering if I should pull the last commit from this PR for now. To be honest with you the flow control in this thing is completely opaque to me at the moment.

Reply:
Same here, the flow control here always leaves me feeling defeated.